src/tx/applyTx/checkPeriodVote.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2018-present, Leap DAO (leapdao.org)
+ *
+ * This source code is licensed under the Mozilla Public License Version 2.0
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+const { Type } = require('leap-core');
+const { bufferToHex } = require('ethereumjs-util');
+
+// const checkEnoughVotes = require('../../period/utils/checkEnoughVotes');
+// const submitPeriod = require('../../txHelpers/submitPeriod');
+// const { logTx } = require('../../utils/debug');
+
+module.exports = async (state, tx) => {
+  if (tx.type !== Type.PERIOD_VOTE) {
+    throw new Error('[period vote] periodVote tx expected');
+  }
+
+  const { slotId } = tx.options;
+
+  if (!state.slots[slotId]) {
+    throw new Error(`[period vote] Slot ${slotId} is empty`);
+  }
+
+  if (
+    !tx.inputs ||
+    !tx.inputs[0].signer ||
+    tx.inputs[0].signer !== state.slots[slotId].signerAddr
+  ) {
+    throw new Error(
+      `[period vote] Input should be signed by validator: ${state.slots[slotId].signerAddr}`
+    );
+  }
+
+  const periodRoot = bufferToHex(tx.inputs[0].prevout.hash);
```

Comment:
also make sure that `tx.inputs[0].prevout.pos == 0`, otherwise it would screw up the index later.